morebits.js

```diff
@@ -2406,6 +2412,44 @@ Morebits.wiki.page = function(pageName, currentAction) {
 		}
 	};
 
+	/**
+	 * Undeletes a page (for admins only)
+	 * @param {Function} onSuccess - callback function to run on success
+	 * @param {Function} [onFailure] - callback function to run on failure (optional)
+	 */
+	this.undeletePage = function(onSuccess, onFailure) {
+		ctx.onUndeleteSuccess = onSuccess;
+		ctx.onUndeleteFailure = onFailure || emptyFunction;
+
+		// if a non-admin tries to do this, don't bother
+		if (!Morebits.userIsInGroup('sysop')) {
```

Comment:
This would unnecessarily slow down the code in batchundelete. Better to put this in `fnProcessUndeleteError`. No decently written script would show an undelete button to non-admins anyway.

This would apply to deletePage function also.

Reply:
Isn't this backwards?  Failing here means non-sysop don't do the later protection query.  At any rate, it's quite quick as is (although admittedly slowing things down isn't bad in our current situation).

Reply:
To clarify, this would slow down the code _for sysops_. Yes, it would speed up things a great deal for non-sysops but what good is that since they won't be able to undelete anything after all?

Reply:
I imagine the impetus was to allow for the "retrieving undelete token..." message